File: backend/app/schemas/stock_request.py
from datetime import datetime

from pydantic import BaseModel


class StockRequestCreate(BaseModel):
    product_id: int
    quantity: int


class StockRequestOut(BaseModel):
    id: int
    product_id: int
    product_name: str
    branch_id: int
    quantity: int
    requested_by: int
    created_at: datetime

File: frontend/src/types/stockRequest.ts
// backend/app/schemas/stock_request.py::StockRequestOut ile birebir eşleşir.
export interface StockRequestItem {
  id: number;
  product_id: number;
  product_name: string;
  branch_id: number;
  quantity: number;
  requested_by: number;
  created_at: string;
}
